pkg/cmd/repo/create/create.go

```diff
@@ -325,11 +313,7 @@ func createRun(opts *CreateOptions) error {
 
 		templateRepoName := opts.Template
 		if !strings.Contains(templateRepoName, "/") {
-			host, err := cfg.DefaultHost()
```

Comment:
Please revert back to reading the DefaultHost. It might not be immediately apparent, but `ghinstance.Default()` isn't the same. The latter is always `github.com`, while the former has awareness of GitHub Enterprise configuration.